proposals/0048-precompile-for-secp256r1-sigverify.md

```diff
@@ -0,0 +1,274 @@
+- The following struct serialized, for each signature to verify
+
+```rust
+struct Secp256r1SignatureOffsets {
+    signature_offset: u16,             // offset to secp256r1 signature of 64 bytes
+    signature_instruction_index: u16,  // instruction index to find signature
+    public_key_offset: u16,            // offset to compressed public key of 33 bytes
+    public_key_instruction_index: u16, // instruction index to find public key
+    message_data_offset: u16,          // offset to start of message data
+    message_data_size: u16,            // size of message data
+    message_instruction_index: u16,    // index of instruction data to get msg data
+}
+```
+
+Multiple signatures can be verified. If any of the signatures fail to verify,
+an error is returned.
+
+The program logic will be constructed and built using a `verify`
+function, as outlined in
+[sdk/src/precompiles.rs](https://github.com/solana-labs/solana/blob/9ffbe2afd8ab5b972c4ad87d758866a3e1bb87fb/sdk/src/precompiles.rs).
+
+Apart from the signature verification, the remaining
+logic should be constructed analogously to the existing
+[ed25519](https://github.com/solana-labs/solana/blob/master/sdk/src/ed25519_instruction.rs)
+& [secp256k1](https://github.com/solana-labs/solana/blob/9ffbe2afd8ab5b972c4ad87d758866a3e1bb87fb/sdk/src/secp256k1_instruction.rs#L4)
+precompiles.
+
+### Implementation
+
+The precompile can be implemented using the `p256` crate at version `0.10.1`.
+This crate is part of the `Rust Crypto` library and implements the NIST P-256
+curve as well as ECDSA. It conforms with the test vectors found in
+[RFC6979](https://datatracker.ietf.org/doc/html/rfc6979#appendix-A.2.5).
+
+- `p256::ecdsa::VerifyingKey::from_sec1_bytes()`
+- `p256::ecdsa::Signature::from_scalars()`
+- `p256::arithmetic::Scalar::is_high()`
+- `p256::ecdsa::VerifyingKey::verify()`
+
+Note: The crate is well maintained, but has never been externally audited.
+
+### Compute Cost / Efficiency
+
+Once the implementation is finished, benchmarking should take place on a
+sufficiently powerful machine in order to determine average compute time per
+signature. Calculation of CUs would be based on the 1 CU / ns convention.
+The secp256k1 ecrecover syscall, which incurs a cost of 25_000 CUs, can be used
+as a reference point.
+
+This is in line with how previous precompiles for EC group operations and
+arithmetic were evaluated/benchmarked.
+See [PR#27961](https://github.com/solana-labs/solana/pull/27961) & [PR#28503](https://github.com/solana-labs/solana/pull/28503)
+
+## Impact
+
+Would enable the on-chain usage of Passkeys and the WebAuthn Standard, and
+turn the vast majority of modern smartphones into native hardware wallets.
+
+By extension, this would also enable the creation of account abstractions and
+forms of Two-Factor Authentication around those keypairs.
+
+## Security Considerations
+
+As [Firedancer](https://github.com/firedancer-io/firedancer) is being developed
```

Comment:
Please remove Firedancer and any mentions by name.
SIMDs should be a timeless document without any implementation specifics.
The compatibility risks are present to anyone who wants to develop a client. 
It would be great to list out potential risk areas (e.g. handling of invalid points). If further research is needed, that should be mentioned too.

Reply:
Removed any mentions of firedancer 👍 
With regard to the risk areas, shall I add that as a subsection in the `Security Considerations` ?